### PR TITLE
Homework6

### PR DESCRIPTION
**Pull request by Artemzhi38**

### CODE REVIEW

File: homeworks/homework6/task1.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    original_init = cls.__init__
+
+    def __custom_init__(*args, **kwargs):
+        """"Adds to original init-function incrementing of
+        nonlocal instance variable value"""
+        original_init(*args, **kwargs)
+        nonlocal instance
+        instance += 1
+
+    def get_created_instances(_=None):
```
Comment (icanhardcode) [inline homeworks/homework6/task1.py]: wtf? if u dont need any params - dont write anything

File: homeworks/homework6/task2.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    def check_homework(self, homework_result: HomeworkResult):
+        if len(homework_result.solution) >= 5:
+            self.homework_done[homework_result.homework].add(homework_result)
+            return True
+        else:
+            return False
+
+    @classmethod
+    def reset_results(cls, homework=None):
+        if homework is None:
```
Comment (icanhardcode) [inline homeworks/homework6/task2.py]: remember how to write conditions in python

**Review by icanhardcode — CHANGES_REQUESTED** [review]

File: homeworks/homework6/task1.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+возвращает значение до сброса
+Имя декоратора и методов не менять
+Ниже пример использования
+"""
+
+
+def instances_counter(cls):
+    """Decorates any class with get_created_instances
+    and reset_instances_counter methods"""
+    instance = 0
```
Comment (icanhardcode) [inline homeworks/homework6/task1.py]: should be a class attribute (static)

File: homeworks/homework6/task1.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    original_init = cls.__init__
+
+    def __custom_init__(self, *args, **kwargs):
+        """"Adds to original init-function incrementing of
+        nonlocal instance variable value"""
+        original_init(self, *args, **kwargs)
+        nonlocal instance
+        instance += 1
+        setattr(self, 'get_created_instances', get_created_instances)
+        setattr(self, 'reset_instances_counter', reset_instances_counter)
```
Comment (nikitosboroda) [inline homeworks/homework6/task1.py:25]: wrong place

File: homeworks/homework6/task2.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        reset_results - если передать экземпряр Homework - удаляет только
+        результаты этого задания из homework_done, если ничего не передавать,
+        то полностью обнулит homework_done."""
+    homework_done = defaultdict(set)
+
+    def check_homework(self, homework_result: HomeworkResult):
+        if len(homework_result.solution) >= 5:
+            self.homework_done[homework_result.homework].add(homework_result)
+            return True
+        else:
```
Comment (icanhardcode) [inline homeworks/homework6/task2.py]: u dont need to write ```else``` if u do ```return``` in ```if``` statement

**Review by icanhardcode — CHANGES_REQUESTED** [review]

File: homeworks/homework6/task1.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    setattr(cls, 'instance', 0)
+    original_init = cls.__init__
+
+    def __custom_init__(self, *args, **kwargs):
+        """"Adds to original init-function incrementing of
+        nonlocal instance variable value"""
+        original_init(self, *args, **kwargs)
+        cls.instance += 1
+        setattr(self, 'get_created_instances', get_created_instances)
+        setattr(self, 'reset_instances_counter', reset_instances_counter)
```
Comment (icanhardcode) [inline homeworks/homework6/task1.py]: look at this

File: homeworks/homework6/task1.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    def reset_instances_counter():
+        """сбрасывает счетчик экземпляров, возвращает значение до сброса"""
+        value_before_reset = cls.instance
+        cls.instance = 0
+        return value_before_reset
+
+    cls.__init__ = __custom_init__
+    setattr(cls, 'get_created_instances', get_created_instances)
+    setattr(cls, 'reset_instances_counter', reset_instances_counter)
```
Comment (icanhardcode) [inline homeworks/homework6/task1.py:38]: and look at this... these rows are the same.. left only one pair of these.. make a right choice

**Review by icanhardcode — CHANGES_REQUESTED** [review]

**Review by icanhardcode — APPROVED** [review]

**Pull request merged**